crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -343,81 +348,34 @@ impl SettlementEncoder {
             }
         };
 
-        sell_amount
-            .checked_mul(uniform_sell_price)
-            .context("custom_buy_price computation failed")?
-            .checked_div(buy_amount)
-            .context("custom_buy_price computation failed")
-    }
-
-    /// Uses either the existing UCP for the sell token or the sell price within the order to
-    /// compute the price of the buy token right at the limit price of the passed order.
-    /// This is needed to settle liquidity orders in a way that gives them no surplus.
-    fn compute_limit_buy_price(&self, order: &Order) -> Result<U256> {
-        // Liquidity orders are settled at their limit price. We set:
-        // buy_price = sell_price * order.sellAmount / order.buyAmount, where sell_price is given from uniform clearing prices
-
-        // Rounding error checks:
-        // Following limit price constraint is checked in the smart contract:
-        // order.sellAmount.mul(sellPrice) >= order.buyAmount.mul(buyPrice),
-
-        // This equation will always be true, as the following equivalence is showing
-        // order.sellAmount.mul(sellPrice)  >= order.sellAmount.mul(sellPrice) holds, as the left and ride side are equal.
-        // Dividing first and then multiplying by order.buyAmount can only make the right side smaller
-        // <=> order.sellAmount.mul(sellPrice)  >= order.buyAmount.mul(sell_price * order.sellAmount / order.buyAmount)
-        // <=> order.sellAmount.mul(sellPrice)  >= order.buyAmount.mul(buyPrice)
-        // <=> equation from smart contract
-
-        self.clearing_prices
-            .get(&order.data.sell_token)
-            .unwrap_or(&order.data.buy_amount)
-            .checked_mul(order.data.sell_amount)
-            .context("buy_price calculation failed")?
-            .checked_div(order.data.buy_amount)
-            .context("buy_price calculation failed")
+        let adjusted_sell_price = buy_amount;
+        let adjusted_buy_price = sell_amount;
```

Comment:
This always confuses me 🤣 - but I'm 99% sure its correct (you can never be 100% when it comes to prices and exchange rates...)